### PR TITLE
UCS/ARRAY/STRING: Add on-stack constructors for array and string buffer

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucs/datastruct/string_buffer.h [hunk 1]
```diff
@@ -35,10 +62,13 @@ UCS_ARRAY_DECLARE_TYPE(string_buffer, size_t, char)
...
-#define UCS_STRING_BUFFER_FIXED(_var, _buffer) \
-    ucs_string_buffer_t _var = { \
-        UCS_ARRAY_FIXED_INITIALIZER((_buffer), ucs_static_array_size(_buffer)) \
-    }
+#define UCS_STRING_BUFFER_STATIC(_var, _buffer) \
+    UCS_STRING_BUFFER_FIXED(_var, _buffer, ucs_static_array_size(_buffer))
+
+
+#define UCS_STRING_BUFFER_ONSTACK(_var, _capacity) \
+    UCS_STRING_BUFFER_FIXED(_var, UCS_ARRAY_ALLOC_ONSTACK(string_buffer, _capacity), \
```
Comment (hoopoepg) [inline src/ucs/datastruct/string_buffer.h:70]: there could be issue in compilation: macro ```UCS_ALLOCA_MAX_SIZE``` is defined in compiler.h file which is not installed with package, but current file (and some other files) are installed (listed in section ```nobase_dist_libucs_la_HEADERS``` of makefile.am)

Comment (hoopoepg) [inline src/ucs/datastruct/string_buffer.h:70]: UCS_ALLOCA_MAX_SIZE is moved.... missed this :)

**Review by hoopoepg — APPROVED** [review]

File: src/ucs/datastruct/array.h [hunk 2]
```diff
@@ -133,25 +127,54 @@ BEGIN_C_DECLS
...
+        (_array)->buffer   = NULL; \
+        (_array)->length   = 0; \
+        (_array)->capacity = 0; \
+    }
+
+
+/**
+ * Initialize a fixed-size array with existing buffer as backing storage.
+ *
+ * @param _name      Array name
```
Comment (dmitrygx) [inline src/ucs/datastruct/array.h:155]: need to remove `_name`

File: src/ucs/datastruct/string_buffer.h [hunk 3]
```diff
@@ -18,6 +18,33 @@ BEGIN_C_DECLS
...
+ * {
+ *    UCS_STRING_BUFFER_FIXED(strb, buffer, max_legth);
+ *    ucs_string_buffer_appendf(&strb, "%x%x", 57005, 48879);
+ *    return buffer;
+ * }
+ * @endcode
+ */
+#define UCS_STRING_BUFFER_FIXED(_var, _buffer, _capacity) \
+    ucs_string_buffer_t _var = { \
+        UCS_ARRAY_FIXED_INITIALIZER((_buffer), _capacity) \
```
Comment (dmitrygx) [inline src/ucs/datastruct/string_buffer.h:44]: no need for brackets around `buffer`

**Review by dmitrygx — APPROVED** [review]

**Review by dmitrygx — APPROVED** [review]

**Review by hoopoepg — APPROVED** [review]

**Pull request merged**